Modify rule S2631(Common): Update text to Education Framework structure (APPSEC-125) (#1218)
SonarSource/rspec: rules/S2631/common/fix/validation.adoc

==== Sanitization and validation

> 
> The easiest way to protect against regex injections is to escape metacharacters and escape sequences, preferably
> with native functions.
> 

Author: ```suggestion
The easiest way to protect against regex injections is to escape metacharacters,
 preferably with native functions. +
The escape function instructs the regular expression engine to interpret these
characters literally and not as metacharacters.
```

==== Sanitization and validation

> As a rule of thumb, the best approach to protecting against injections is to

Author: agreed 👍

> As a rule of thumb, the best approach to protecting against injections is to

Author: ```suggestion
```

> systematically ensure that untrusted data cannot break out of its initially
> intended logic.

Author: ```suggestion
```

For regex pattern injections, the easiest way to protect against this

> vulnerability is to escape metacharacters and escapes sequences, preferably

Author: yeah, I'd say "regex pattern" and "regex expression" are the same

with native functions.

> Another clean method is to validate the input before using it to scan a string.

Reviewer: I think we can remove this line and introduce the following with something like, "As an alternative..." and refer to it as an allow-list approach.

> Another clean method is to validate the input before using it to scan a string.

Author: Yup. I'm deleting it and adding the "alternative" expression to the next paragraph.

> Another clean method is to validate the input before using it to scan a string.

Author: ```suggestion
```

> Create a list of authorized and secure strings that you want the application to

Author: ```suggestion
As an alternative, create a list of authorized and secure strings that you want the application to
```

be able to use in a query. +
If a user input does not match an entry in this list, it should be rejected
because it is considered unsafe.

The list can be either a regex string, an array, or validators on specific
ranges of characters. If you use regexes, choose simple regexes to avoid ReDOS

> attacks.

Reviewer: I think this is not very clear, and it appears to add some slighlty  duplicated content namely on the "validators on specific ranges of characters", maybe we can remove it.

Author: agreed 

```suggestion
```
